### PR TITLE
Remove assumptions based autosimplifications from Pow

### PR DESCRIPTION
**Pull request by debugger22**

### CODE REVIEW

**Comment by debugger22** [comment]
Following two tests give wrong results. But they work as expected with refine.

```
assert ((-k)**n).is_nonnegative is None
assert ((-k)**m).is_nonpositive is None
```

**Comment by asmeurer** [comment]
What are the assumptions on `k` and `n`?

**Comment by debugger22** [comment]
```
     k = Symbol('k', integer=True, nonnegative=True)
     n = Symbol('n', even=True)
     m = Symbol('m', odd=True)
```

**Comment by asmeurer** [comment]
Why do those tests test that they give None? It seems that True is correct in each case.

**Comment by debugger22** [comment]
Here's the commit which altered the returned values from `True` to `None`. https://github.com/sympy/sympy/commit/58651bb78a80426fbd46b45e200a2ab539198b1b

**Comment by asmeurer** [comment]
I guess that was a mistake. The assumptions regressed, but you should never test that an assumption gives None unless it should be None (the tests should have been put in an XFAIL test). 

So the changes here are definitely fine. I don't see why the simplified forms shouldn't be able to give True as well, though.

**Comment by debugger22** [comment]
Ok. I'm removing the XFAIL.

**Comment by debugger22** [comment]
@asmeurer Please review.

**Comment by asmeurer** [comment]
Shouldn't use is_Symbol here too, like with exp?

**Comment by debugger22** [comment]
Yep. Consistency.

**Comment by debugger22** [comment]
@asmeurer this is ready.

File: sympy/core/power.py [hunk 1]
```diff
@@ -162,7 +162,8 @@ def __new__(cls, b, e, evaluate=None):
                 return S.One
             elif e is S.One:
                 return b
-            elif e.is_integer and _coeff_isneg(b):
+            # Only perform autosimplification if base if a Symbol or number
+            elif (b.is_Symbol or b.is_number) and e.is_integer and _coeff_isneg(b):
```
Comment (asmeurer) [inline sympy/core/power.py]: Why do you only check `b`?

Comment (debugger22) [inline sympy/core/power.py]: I've assumed "don't care" on exponent. Should I put checks on exponent as well associated with `and`?

Comment (asmeurer) [inline sympy/core/power.py]: What do you mean "don't care"?

Comment (debugger22) [inline sympy/core/power.py]: It can be anything. Number, Symbol, Expr, etc.

Comment (asmeurer) [inline sympy/core/power.py]: Eventually everything in the core should only look at assumptions if it's a number or Symbol, but we are doing things in parts. Does too much break if you apply the rule to `e`?

**Comment by debugger22** [comment]
@asmeurer updated this one. Please review.

**Comment by debugger22** [comment]
@asmeurer I think it's complete. Do you have any suggestion for it?

**Comment by asmeurer** [comment]
Sorry for sitting on this. I didn't have much time for SymPy but I will have much more time now. This looks good. Let's merge it.

**Comment by debugger22** [comment]
Thanks for merging it! I didn't see your message until today.

File: sympy/core/tests/test_eval_power.py [hunk 2]
```diff
@@ -336,7 +334,7 @@ def test_issue_7638():
     assert sqrt(e**6) == e**3
     assert sqrt((1 + I*r)**6) != (1 + I*r)**3
 
-
+@XFAIL
```
Comment (gxyd) [inline sympy/core/tests/test_eval_power.py]: I was wondering that why were these `@XFAIL` added? I think these pass on master.

**Pull request merged**